generator/rangewriter.go

```diff
@@ -48,15 +57,30 @@ func (rw *RangeWriter) WriteStringLiteral(level int, s string) (r parser.Range,
 		if err != nil {
 			return
 		}
-		if _, err = rw.WriteIndent(level, `_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString("`); err != nil {
-			return
+
+		if rw.extractStrings {
+			index := len(rw.strings)
+			if _, err = rw.WriteIndent(level, "templ_7745c5c3_Err = templ.WriteExtractedString(templ_7745c5c3_Buffer, "+strconv.Itoa(index)+")\n"); err != nil {
+				return
+			}
+		} else {
+			if _, err = rw.WriteIndent(level, `_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString("`); err != nil {
+				return
+			}
 		}
 	}
-	_, err = rw.write(s)
-	if err != nil {
-		return
+
+	if rw.extractStrings {
+		rw.strings[len(rw.strings)-1] += s
```

Comment:
Ah, now I see why the strings slice is initialised with an empty string. I wonder now whether having a slice of strings.Builder would be more efficient in terms of memory allocations.

I think it might be better.

As you likely know, concatenating strings causes a new string to be created that's the size of both strings.

In small strings, the cost of initialising the string builder can be higher.

Reply:
Used a `strings.Builder` and got rid of the slice and string concatenation.

Reply:
But I use only one builder and just add a new line for a new literal so there is no extra cost